### PR TITLE
Moved method getRouteKey(msg) from AuthorizationRouter and ProfileRou…

### PR DESCRIPTION
**Pull request by Alexey007gold**
…ter to new abstract class AbstractByCommandRouterHandler.

### CODE REVIEW

File: basis/common-router/src/main/java/com/alexkoveckiy/common/router/api/AbstractByCommandRouterHandler.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.alexkoveckiy.common.protocol.Request;
+
+/**
+ * Created by alex on 03.03.17.
+ */
+public abstract class AbstractByCommandRouterHandler<T extends Handler> extends AbstractRouterHandler<T> {
+
+    @Override
+    public String getRouteKey(Request<?> msg) {
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alexkoveckiy/common/router/api/AbstractByCommandRouterHandler.java]: @Alexey007gold You've implemented Factory. So this method must be in factory.

File: basis/common-router/src/main/java/com/alexkoveckiy/common/router/impl/FirstRouter.java [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
 /**
  * Created by alex on 25.02.17.
  */
 
 @Component
-public class FirstRouter extends AbstractRouterHandler<CommonRouterHandler> {
+public class FirstRouter implements RouterHandler {
+
+    @Autowired
+    private HandlerFactory handlerFactoryByType;
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alexkoveckiy/common/router/impl/FirstRouter.java]: @Alexey007gold How your factory will know wick handlers to use?

File: basis/common-router/src/main/java/com/alexkoveckiy/common/router/impl/FirstRouter.java [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
 
     @Override
     public String getName() {
         return "first_router";
     }
 
     @Override
-    public String getRouteKey(Request<?> msg) {
-        return msg.getHeader().getType();
+    public Response<?> handle(Request<?> msg) {
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alexkoveckiy/common/router/impl/FirstRouter.java]: @Alexey007gold all this logic must be in abstract class

File: services/authorization/authorization-impl/src/main/java/com/alexkoveckiy/authorization/impl/router/AuthorizationRouter.java [hunk 4]
```diff
@@ -1,23 +1,19 @@
...
+import com.alexkoveckiy.common.router.api.RequestHandler;
 import org.springframework.stereotype.Component;
 
 /**
  * Created by alex on 23.02.17.
  */
 
 @Component
-public class AuthorizationRouter extends AbstractRouterHandler<AuthorizationRouterHandler> {
+public class AuthorizationRouter extends AbstractRouterHandler<RequestHandler> implements CommonRouterHandler {
```
Comment (Alexey007gold) [inline services/authorization/authorization-impl/src/main/java/com/alexkoveckiy/authorization/impl/router/AuthorizationRouter.java]: When i change it to more concrete like: extends AbstractRouterHandler\<AuthorizationRequestHandler\>, then Spring says that there is 2 autowire candidate for HandlerFactory

Comment (Dominys) [reply services/authorization/authorization-impl/src/main/java/com/alexkoveckiy/authorization/impl/router/AuthorizationRouter.java]: @Alexey007gold Than think how to solve this.

**Review by Dominys — APPROVED** [review]

**Comment by Dominys** [comment]
@Alexey007gold resolve conflicts

**Pull request merged**